File: rubyPreExamen/lib/PowerEfficientSpaceStation.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'SpaceStation'
require_relative 'PowerEfficientSpaceStationToUI'
require_relative 'SuppliesPackage'

module Deepspace
class PowerEfficientSpaceStation < SpaceStation
  
  @@EFFICIENCYFACTOR = 1.10
  
  def initialize(station)
    s = SuppliesPackage.new(station.ammoPower, station.fuelUnits, station.shieldPower)
    super(station.name, s, station.weapons, station.shieldBoosters,
      station.nMedals)
  end
  
  def self.newCopy(station)
    h = super
    return h
  end
  
  def setLoot(loot)
    super
    return Transformation::NOTRANSFORM
  end
  
  def fire
    shot = super
    shot*@@EFFICIENCYFACTOR
  end
  
  def protection
    shield = super
    shield*@@EFFICIENCYFACTOR
  end
  
  def getUIversion
    return PowerEfficientSpaceStation.new(self)
  end
end
end

File: javaSuyo/src/deepspace/PowerEfficientSpaceStation.java
/*
 * Here comes the text of your license
 * Each line should be prefixed with  * 
 */
package deepspace;

/**
 *
 * @author yabir
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    
    private static final float EFFICIENCYFACTOR = 1.10f;
    
    PowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot l){
        if(super.setLoot(l)==Transformation.GETEFFICIENT)
            return Transformation.GETEFFICIENT;
        else
            return Transformation.NOTRANSFORM;
    }
    
    @Override
    public PowerEfficientSpaceStationToUI getUIversion(){
        return new PowerEfficientSpaceStationToUI(this);
    }
    
}
